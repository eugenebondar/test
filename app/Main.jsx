import Routes from './Router';
import ReactDOM from 'react-dom';

import './Main.less';

import 'babel-polyfill';

ReactDOM.render(Routes, document.querySelector('.container'));
