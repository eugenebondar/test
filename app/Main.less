@import 'grid';
@import 'MainTheme';

* {
  box-sizing: border-box;
}

html,
body {
  height: 100%;
}

body {
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  font-weight: 300;
  margin: 0;
  padding: 0;
  height: 100%;
}

ul, ol {
  list-style: none;
  margin: 0;
  padding: 0;
}

img {
  max-width: 100%;
}

h1 {
  font-size: 36px;
  font-weight: 300;
  margin-bottom: 35px;
}

.footer {
  background-color: #f2f2f2;
  flex: 0 0 auto;
  padding: 15px 0;
}

.footer li {
  display: inline-block;
  margin-right: 15px;
}
.footer li a {
  color: #222222;
  opacity: 0.6;
  text-decoration: none;
}
.footer li a:hover {
  text-decoration: underline;
}

.app__wrapper, .container {
  display: flex;
  flex-direction: column;
  height: 100%;
}

.main__wrap {
  flex: 1 0 auto;
}
.app__content {
  padding: 25px 0;
}
