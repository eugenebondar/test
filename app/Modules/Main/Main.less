.main {
  //todo add here some styles or remove
}
