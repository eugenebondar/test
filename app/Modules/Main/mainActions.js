import Reflux from 'reflux';

const mainActions = {
    init: Reflux.createAction({ asyncResult: true }),
    destroy: Reflux.createAction({ sync: true })
};

const text = 'this is data coming from action';

mainActions.init.listen(function() {
    (Math.random() > 0.5) ? this.completed({ text }) : this.failed();
});

export default mainActions;
