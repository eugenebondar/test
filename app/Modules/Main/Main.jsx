import React from 'react';
import classNames from 'classnames';
import './Main.less'; //todo add there some styles or remove

export default class Main extends React.PureComponent {

    static propTypes = {
        className: React.PropTypes.string,
        isError: React.PropTypes.bool,
        isLoading: React.PropTypes.bool,
        text: React.PropTypes.string
    };

    static defaultProps = {
        className : ''
    };

    render() {
        const finalClassName = classNames('main', this.props.className);
        return <div className={finalClassName}>
            Hello world
            {this.props.isError && <div>There was error</div>}
            {this.props.isLoading && <div>Loading</div>}
            <div>{this.props.text}</div>
        </div>;
    }
}
