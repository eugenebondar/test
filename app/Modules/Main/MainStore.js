import Reflux from 'reflux';
import mainActions from './mainActions';

export const defaultState = {
    isLoading: false,
    isError: false
};

class MainStore extends Reflux.Store {

    constructor() {
        super();

        this.listenables = mainActions;

        this.state = { ...defaultState };
    }

    /**
     * @listens mainActions.init
     */
    onInit() {
        this.setState({
            isLoading: true
        });
    }

    /**
     * @listens mainActions.init.completed
     */
    onInitCompleted({ text }) {
        this.setState({
            text,
            isLoading: false
        });
    }

    /**
     * @listens mainActions.init.failed
     */
    onInitFailed() {
        this.setState({
            isLoading: false,
            isError: true
        });
    }

    /**
     * @listens mainActions.destroy
     */
    onDestroy() {
        this.setState(defaultState);
    }
}

export default Reflux.initStore(MainStore);
