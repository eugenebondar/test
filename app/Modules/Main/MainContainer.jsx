import { Component } from 'reflux';
import React from 'react';
import MainStore from './MainStore';
import Main from './Main';
import mainActions from './mainActions';

export default class MainContainer extends Component.extend(React.PureComponent) {

    static propTypes = {};

    static defaultProps = {};

    constructor() {
        super();

        this.store = MainStore;

        this.mapStoreToState(MainStore, () => {
            //todo implement logic here or remove it
        });
    }

    componentDidMount() {
        mainActions.init();
    }

    componentWillUnmount() {
        super.componentWillUnmount();
        mainActions.destroy();
    }

    render() {
        return <Main
            {...this.state}
        />;
    }
}
