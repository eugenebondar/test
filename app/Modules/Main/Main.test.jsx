import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Main from './Main';

describe('Main', () => {
    describe('with default set of props', () => {
        let component, errorConsole;

        beforeAll(() => {
            errorConsole = spyOn(console, 'error');
        });

        beforeEach(() => {
            component = shallow(<Main/>);
        });

        it('matches snapshot', () => {
            expect(shallowToJson(component)).toMatchSnapshot();
        });

        it('prints no errors from propTypes validation', () => {
            expect(errorConsole).not.toHaveBeenCalled();
        })
    })
});
