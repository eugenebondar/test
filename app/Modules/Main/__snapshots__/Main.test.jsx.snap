exports[`Main with default set of props matches snapshot 1`] = `
<div
  className="main">
  Hello world
  <div />
</div>
`;
